x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/remoteaccess/RemoteAccessHeadersForCcsRestIT.java

```diff
@@ -0,0 +1,506 @@
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.common.UUIDs.randomBase64UUID;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.is;
+
+public class RemoteAccessHeadersForCcsRestIT extends SecurityOnTrialLicenseRestTestCase {
+    @BeforeClass
+    public static void checkFeatureFlag() {
+        assumeTrue("untrusted remote cluster feature flag must be enabled", TcpTransport.isUntrustedRemoteClusterEnabled());
+    }
+
+    private static final String CLUSTER_A = "my_remote_cluster_a";
+    private static final String CLUSTER_B = "my_remote_cluster_b";
+    private static final String REMOTE_SEARCH_USER = "remote_search_user";
+    private static final SecureString PASSWORD = new SecureString("super-secret-password".toCharArray());
+    private static final String REMOTE_SEARCH_ROLE = "remote_search";
+
+    private final ThreadPool threadPool = new TestThreadPool(getClass().getName());
+
+    @Before
+    public void setup() throws IOException {
+        createUser(REMOTE_SEARCH_USER, PASSWORD, List.of(REMOTE_SEARCH_ROLE));
+
+        final var putRoleRequest = new Request("PUT", "/_security/role/" + REMOTE_SEARCH_ROLE);
+        putRoleRequest.setJsonEntity("""
+            {
+              "indices": [
+                {
+                  "names": ["index-a"],
+                  "privileges": ["read"]
+                }
+              ],
+              "remote_indices": [
+                {
+                  "names": ["index-a"],
+                  "privileges": ["read", "read_cross_cluster"],
+                  "clusters": ["my_remote_cluster*"]
+                },
+                {
+                  "names": ["index-b"],
+                  "privileges": ["read", "read_cross_cluster"],
+                  "clusters": ["my_remote_cluster_b"]
+                }
+              ]
+            }""");
+        assertOK(adminClient().performRequest(putRoleRequest));
+
+        final var indexDocRequest = new Request("POST", "/index-a/_doc?refresh=true");
+        indexDocRequest.setJsonEntity("{\"foo\": \"bar\"}");
+        assertOK(adminClient().performRequest(indexDocRequest));
+    }
+
+    @After
+    public void cleanup() throws IOException {
+        deleteUser(REMOTE_SEARCH_USER);
+        deleteRole(REMOTE_SEARCH_ROLE);
+        deleteIndex(adminClient(), "index-a");
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+        ThreadPool.terminate(threadPool, 10, TimeUnit.SECONDS);
+    }
+
+    public void testRemoteAccessHeadersSentSingleRemote() throws Exception {
+        final BlockingQueue<CapturedActionWithHeaders> capturedHeaders = ConcurrentCollections.newBlockingQueue();
+        try (MockTransportService remoteTransport = startTransport("remoteNodeA", threadPool, capturedHeaders)) {
```

Comment:
TIL. This is a clever way for cross cluster testing.

Reply:
++